core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -183,6 +183,10 @@ public String indexedShapeType() {
      * @return this
      */
     public GeoShapeQueryBuilder strategy(SpatialStrategy strategy) {
+        if (strategy != null && strategy == SpatialStrategy.TERM && relation != ShapeRelation.INTERSECTS) {
```

Comment:
The default for `strategy` currently is to be unset (`null` that is), in which case the field types default strategy is going to be used later when building the lucene query. That's why I thought it's okay if the setter is used with `null` value as well, although it would be a no-op. Might be even useful if `strategy` was set to something else and needs to be changed back to the default.